### ISSUE TITLE
Module "dev.gitlive:firebase-auth" has a reference to symbol dev.gitlive.firebase.auth/FirebaseUser.displayName..

### ISSUE THREAD

**Issue by OmarShehe** [GitLiveApp/firebase-kotlin-sdk#296]
Using `dev.gitlive:firebase-auth:1.6.1`
I'm getting this error when trying to access ` Firebase.auth`

```
This could happen if the required dependency is missing in the project. Or if there is a dependency of "dev.gitlive:firebase-auth" that has a different version in the project than the version that "dev.gitlive:firebase-auth" was initially compiled with. Please check that the project configuration is correct and has consistent versions of all required dependencies.

The list of "dev.gitlive:firebase-auth" dependencies that may lead to conflicts:
1. "kotlin" (a library with unknown version)
2. "dev.gitlive:firebase-app" (a library with unknown version)
3. "dev.gitlive:firebase-common" (a library with unknown version)
4. "org.jetbrains.kotlinx:atomicfu" (a library with unknown version)
5. "org.jetbrains.kotlinx:kotlinx-coroutines-core" (a library with unknown version)
6. "org.jetbrains.kotlinx:kotlinx-serialization-core" (a library with unknown version)

Project dependencies:
+--- kotlin
+--- androidx.compose.runtime:runtime
|    +--- kotlin
|    +--- org.jetbrains.kotlinx:atomicfu
|    |    \--- kotlin
|    \--- org.jetbrains.kotlinx:kotlinx-coroutines-core
|         +--- kotlin
|         \--- org.jetbrains.kotlinx:atomicfu (*)
+--- app.softwork:kotlinx-uuid-core
|    +--- kotlin
|    \--- org.jetbrains.kotlinx:kotlinx-serialization-core
|         \--- kotlin
+--- app.softwork:routing-compose
|    +--- kotlin
|    +--- androidx.compose.runtime:runtime (*)
|    +--- app.softwork:kotlinx-uuid-core (*)
|    +--- org.jetbrains.compose.web:internal-web-core-runtime
|    |    +--- kotlin
|    |    +--- androidx.compose.runtime:runtime (*)
|    |    +--- org.jetbrains.kotlinx:atomicfu (*)
|    |    \--- org.jetbrains.kotlinx:kotlinx-coroutines-core (*)
|    +--- org.jetbrains.compose.web:web-core
|    |    +--- kotlin
|    |    +--- androidx.compose.runtime:runtime (*)
|    |    +--- org.jetbrains.compose.web:internal-web-core-runtime (*)
|    |    +--- org.jetbrains.kotlinx:atomicfu (*)
|    |    \--- org.jetbrains.kotlinx:kotlinx-coroutines-core (*)
|    +--- org.jetbrains.kotlinx:atomicfu (*)
|    +--- org.jetbrains.kotlinx:kotlinx-coroutines-core (*)
|    \--- org.jetbrains.kotlinx:kotlinx-serialization-core (*)
+--- com.spatialcollective:Microtask
|    +--- kotlin
|    +--- androidx.compose.runtime:runtime (*)
|    +--- app.softwork:kotlinx-uuid-core (*)
|    +--- app.softwork:routing-compose (*)
|    +--- dev.gitlive:firebase-app
|    |    +--- kotlin
|    |    +--- dev.gitlive:firebase-common
|    |    |    +--- kotlin
|    |    |    +--- org.jetbrains.kotlinx:atomicfu (*)
|    |    |    +--- org.jetbrains.kotlinx:kotlinx-coroutines-core (*)
|    |    |    \--- org.jetbrains.kotlinx:kotlinx-serialization-core (*)
|    |    +--- org.jetbrains.kotlinx:atomicfu (*)
|    |    +--- org.jetbrains.kotlinx:kotlinx-coroutines-core (*)
|    |    \--- org.jetbrains.kotlinx:kotlinx-serialization-core (*)
|    +--- dev.gitlive:firebase-auth
|    |    ^^^ This module requires symbol dev.gitlive.firebase.auth/FirebaseUser.displayName.<get-displayName>|-7122534302014937473[1].
|    |    +--- kotlin
|    |    +--- dev.gitlive:firebase-app (*)
|    |    +--- dev.gitlive:firebase-common (*)
|    |    +--- org.jetbrains.kotlinx:atomicfu (*)
|    |    +--- org.jetbrains.kotlinx:kotlinx-coroutines-core (*)
|    |    \--- org.jetbrains.kotlinx:kotlinx-serialization-core (*)
|    +--- dev.gitlive:firebase-common (*)
|    +--- org.jetbrains.compose.web:internal-web-core-runtime (*)
|    +--- org.jetbrains.compose.web:web-core (*)
|    +--- org.jetbrains.kotlinx:atomicfu (*)
|    +--- org.jetbrains.kotlinx:kotlinx-coroutines-core (*)
|    \--- org.jetbrains.kotlinx:kotlinx-serialization-core (*)
+--- dev.gitlive:firebase-app (*)
+--- dev.gitlive:firebase-auth (*)
|    ^^^ This module requires symbol dev.gitlive.firebase.auth/FirebaseUser.displayName.<get-displayName>|-7122534302014937473[1].
+--- dev.gitlive:firebase-common (*)
+--- org.jetbrains.compose.web:internal-web-core-runtime (*)
+--- org.jetbrains.compose.web:web-core (*)
+--- org.jetbrains.kotlinx:atomicfu (*)
+--- org.jetbrains.kotlinx:kotlinx-coroutines-core (*)
\--- org.jetbrains.kotlinx:kotlinx-serialization-core (*)

(*) - dependencies omitted (listed previously)


```

**Comment by rocketraman** [comment]
See my comments in https://github.com/GitLiveApp/firebase-kotlin-sdk/pull/277.

**Comment by rocketraman** [comment]
See this comment: https://github.com/GitLiveApp/firebase-kotlin-sdk/pull/277#issuecomment-1092269329

The root cause of this is a bug in the IR compiler: https://youtrack.jetbrains.com/issue/KT-48836.

A (non-backward-compatible) patch for this is https://github.com/innertech/firebase-kotlin-sdk/commit/0358eb46bf7fa0908f68262ad44b502fb87275bf but any existing code that relies on the default values of the patched call will unintentionally set existing values in Firebase to null. If you don't care about backward compatibility, that patch works just fine.

**Comment by Darkos-den** [comment]
I faced the same issue. @rocketraman Need to download the repository and make a local patch to solve this problem?

**Comment by rocketraman** [comment]
@Darkos-den We do have pre-built GitHub packages, if you want to use those: https://github.com/orgs/innertech/packages?repo_name=firebase-kotlin-sdk.

**Comment by rocketraman** [comment]
For anyone tracking this issue, I've published new patched packages for version 1.6.2 that are built on Kotlin 1.7.10 (they contain the changes from https://github.com/suntrix/firebase-kotlin-sdk/tree/kotlin-1-7-10) as well as the patch for this issue (https://youtrack.jetbrains.com/issue/KT-48836).

https://github.com/orgs/innertech/packages?repo_name=firebase-kotlin-sdk

**Comment by jankusy** [comment]
Hi guys. Any update about this? I also have this issue with v1.8.1. on js target platform. Currently I've decided to use https://github.com/orgs/innertech/packages?repo_name=firebase-kotlin-sdk but It would be great to have the newest version of firebase libs.

**Comment by ahmedsalemelzeiny** [comment]
@rocketraman Can you we get a new published version with this fix and v1.8.1

**Comment by rocketraman** [comment]
@jankusy @ahmedsalemelzeiny Looks like 1.8.2 was released which has a [workaround](https://github.com/GitLiveApp/firebase-kotlin-sdk/pull/385) for this issue.

**Comment by nbransby** [comment]
I believe this has been fixed in 1.9.0 - please reopen if something is missing
